src/common/qheader32.cl

```diff
@@ -0,0 +1,24 @@
+//////////////////////////////////////////////////////////////////////////////////////
+//
+// (C) Daniel Strano and the Qrack contributors 2017, 2018. All rights reserved.
+//
+// This is a multithreaded, universal quantum register simulation, allowing
+// (nonphysical) register cloning and direct measurement of probability and
+// phase, to leverage what advantages classical emulation of qubits can have.
+//
+// Licensed under the GNU Lesser General Public License V3.
+// See LICENSE.md in the project root or https://www.gnu.org/licenses/lgpl-3.0.en.html
+// for details.
+
+#define cmplx float2
+#define cmplx2 float4
+#define cmplx4 float8
```

Comment:
Should this be an... error?

Reply:
No, this are intrinsic vector types that are always recognized in OpenCL. We've used them the entire time.

Reply:
To be clear, a `cmplx` type/macro has two float components represented as a vector, hence the `cmplx` vectors have half the number of components of the `floatn` types that are really underneath.